### PR TITLE
Use new analysis engine when using Interpreter

### PR DESCRIPTION
**Pull request by xuorig**
Automatically opt in into new analysis engine when using the interpreter runtime!

### CODE REVIEW

File: spec/graphql/static_validation/rules/variables_are_used_and_defined_spec.rb [hunk 1]
```diff
@@ -71,7 +71,6 @@
       GRAPHQL
     }
 
-    focus
```
Comment (xuorig) [inline spec/graphql/static_validation/rules/variables_are_used_and_defined_spec.rb]: 😨

File: lib/graphql/analysis/ast.rb [hunk 2]
```diff
@@ -13,11 +13,6 @@ module Analysis
     module AST
       module_function
 
-      def use(schema_defn)
```
Comment (xuorig) [inline lib/graphql/analysis/ast.rb]: Moved to `use Interpreter` instead.

**Comment by xuorig** [comment]
Weird tracing test failing with `TESTING_INTERPRETER`, somehow can't reproduce locally 🤔

**Comment by rmosolgo** [comment]
It turns out a bit convenient that these are separate, for migrating github, I did `use GraphQL::Exection::Interpreter` _first_, then migrated the analyzers afterward. So, let's keep them separate to support a more gradual adoption.

**Pull request closed without merging**